src/common/regex.cpp

```diff
@@ -690,4 +690,39 @@ int wxRegEx::Replace(wxString *pattern,
     return m_impl->Replace(pattern, replacement, maxMatches);
 }
 
+wxString wxRegEx::Escape(const wxString& str)
+{
+    wxString strEscaped;
+
+    for (wxString::const_iterator it = str.begin(); it != str.end(); ++it)
+    {
+        if (s_strMetaChars.find(*it) != wxString::npos)
+        {
+            strEscaped += wxT('\\');
+        }
+
+        strEscaped += *it;
+    }
+
+    return strEscaped;
+}
+
+wxString wxRegEx::Unescape(const wxString& str)
+{
+    wxString strUnescaped;
+
+    for (wxString::const_iterator it = str.begin(); it != str.end(); ++it)
+    {
+        if (*it == wxT('\\') && (it + 1) != str.end() &&
+            s_strMetaChars.find(*(it + 1)) != wxString::npos)
```

Comment:
Is this check necessary? We shouldn't have any backslashes before non-special characters in a properly escaped string.

Reply:
This check will unescape any escaped metacharacter, and if non-special character was preceded by a backslash in the given string, it simply wont unescaped because it is not a metacharacter in the first place, and both the backslash and that character will be left as is.